scripts/ci-build-kubernetes.sh

```diff
@@ -92,7 +97,7 @@ main() {
 
         for IMAGE_NAME in "${IMAGES[@]}"; do
             # extract docker image URL form `docker load` output
-            OLD_IMAGE_URL="$(docker load --input "${KUBE_ROOT}/_output/release-images/amd64/${IMAGE_NAME}.tar" | grep -oP '(?<=Loaded image: )[^ ]*')"
+            OLD_IMAGE_URL="$(docker load --input "${KUBE_ROOT}/_output/release-images/amd64/${IMAGE_NAME}.tar" | ${GREP_BINARY} -oP '(?<=Loaded image: )[^ ]*' | head -n 1)"
```

Comment:
in your comment here: https://github.com/kubernetes/kubernetes/pull/113395#issuecomment-1324029704

it seems that registry.k8s.io is the second image returned in the list not the first, does this mean we're going to still use gcr.io for those cases? If that's the case should we instead grep for registry.k8s.io so we select the right one? Not sure if it matters in practice

Reply:
It shouldn't matter so long as our `docker load` and `docker tag` commands refer to the same image.

Ultimately what we're doing here is re-tagging the image that's a part of the k/k build process, and uploading it to our own registry.

Of course we want to monitor upstream k8s evolution in case there are actually multiple, _different_ container images inside a single tar (definitely seems that should never happen).

`head -n 1` seems about the simplest way of doing this. In the case that there's a bunch of differently named images, it'll just pick one (the first, obviously), and in the case where there's only one image, it's essentially a no-op.